python/ray/tune/ray_trial_executor.py

```diff
@@ -384,6 +385,24 @@ def _setup_remote_runner(self, trial):
             kwargs["remote_checkpoint_dir"] = trial.remote_checkpoint_dir
             kwargs["sync_function_tpl"] = trial.sync_function_tpl
 
+            # Throw a meaningful error if trainable does not use the
+            # new API
+            sig = inspect.signature(trial.get_trainable_cls())
+            try:
+                sig.bind_partial(**kwargs)
+            except Exception as e:
+                raise RuntimeError(
```

Comment:
This is primarily an improved error message right?

Reply:
Yes, this is to make sure users with legacy trainable classes trying to use cloud checkpointing are guided towards adjusting their API.